packages/synapse-react-client/src/components/Authentication/OneTimePasswordForm.tsx

```diff
@@ -1,67 +1,152 @@
+  hideReset2FA?: boolean
+  onClickPromptReset2FA?: () => void
+  onClickReset2FA?: () => void
+  twoFactorAuthResetIsSuccess: UseLoginReturn['twoFactorAuthResetIsSuccess']
+  twoFactorAuthResetIsPending: UseLoginReturn['twoFactorAuthResetIsPending']
 }
 
+export const SHOW_RECOVERY_CODE_FORM_BUTTON_TEXT = 'Use a backup code instead'
+export const SHOW_TOTP_FORM_BUTTON_TEXT = 'Use authenticator app instead'
+export const BEGIN_RESET_2FA_BUTTON_TEXT = 'Lost access to your codes?'
+export const SEND_RESET_2FA_EMAIL_BUTTON_TEXT = 'Send Instructions'
+
 /**
  * Component that allows the user to enter a one-time password second authentication factor,
  * such as a timed one-time password (TOTP) generated using a shared secret, or a single-use recovery code.
+ *
+ * The user may also use this UI to begin the process of disabling 2FA on their account.
  * @param props
  * @constructor
  */
 export default function OneTimePasswordForm(props: OneTimePasswordFormProps) {
-  const { step, isLoading, onSubmit, onClickUseBackupCode, onClickUseTOTP } =
-    props
+  const {
+    step,
+    loginIsPending,
+    onSubmit,
+    onClickUseBackupCode,
+    onClickUseTOTP,
+    hideReset2FA,
+    onClickPromptReset2FA,
+    onClickReset2FA,
+    twoFactorAuthResetIsSuccess,
+    twoFactorAuthResetIsPending,
+  } = props
+
+  const { isInExperimentalMode } = useSynapseContext()
+
   return (
     <>
       {step === 'VERIFICATION_CODE' && (
         <>
           <TOTPForm
-            isLoading={isLoading}
+            loginIsPending={loginIsPending}
             onSubmit={totp => {
               onSubmit(totp, 'TOTP')
             }}
           />
           <Link
             align={'center'}
             color={'grey.700'}
-            sx={{ display: 'block', mx: 'auto' }}
+            sx={{ display: 'block', mx: 'auto', my: 2 }}
             onClick={() => {
               onClickUseBackupCode()
             }}
           >
-            Use a backup code instead
+            {SHOW_RECOVERY_CODE_FORM_BUTTON_TEXT}
           </Link>
         </>
       )}
       {step === 'RECOVERY_CODE' && (
         <>
           <RecoveryCodeForm
-            isLoading={isLoading}
+            loginIsPending={loginIsPending}
             onSubmit={recoveryCode => {
               onSubmit(recoveryCode, 'RECOVERY_CODE')
             }}
           />
           <Link
             align={'center'}
             color={'grey.700'}
-            sx={{ display: 'block', mx: 'auto' }}
+            sx={{ display: 'block', mx: 'auto', my: 2 }}
+            onClick={() => {
+              onClickUseTOTP()
+            }}
+          >
+            {SHOW_TOTP_FORM_BUTTON_TEXT}
+          </Link>
+        </>
+      )}
+      {!hideReset2FA &&
+        onClickPromptReset2FA &&
+        onClickReset2FA &&
+        step !== 'DISABLE_2FA_PROMPT' &&
+        // TODO: Remove from experimental mode once feature flag is toggled
+        isInExperimentalMode && (
```

Comment:
This feature won't work unless the feature flag is toggled on the backend, so users in experimental mode will get an error if they click this before we enable the feature.